rs/test_utilities/artifact_pool/src/consensus_pool.rs

```diff
@@ -196,15 +197,35 @@ impl TestConsensusPool {
             no_op_logger(),
             time_source.clone(),
         );
+        let membership = Membership::new(pool.get_cache(), registry_client.clone(), subnet_id);
         TestConsensusPool {
             subnet_id,
             registry_client,
             pool,
             time_source,
             dkg_payload_builder,
+            membership,
         }
     }
 
+    // Utility function to determine the identity of the block maker with the
+    // specified rank at a given height. Panics if this rank does not exist.
+    pub fn get_block_maker_by_rank(&self, height: Height, rank: Rank) -> NodeId {
+        let pool_reader = PoolReader::new(&self.pool);
+        *self
+            .membership
+            .get_nodes(height)
+            .unwrap()
+            .iter()
+            .find(|node| {
+                let prev_beacon = pool_reader.get_random_beacon(height.decrement()).unwrap();
```

Comment:
nit: can we move this line out of the closure?

Reply:
Done